src/app/bilinearity-test/bilinearity.ml

```diff
@@ -0,0 +1,35 @@
+open Core
+
+(* 
+   Check
+
+   e(p, a x + b y) = a e(p, x) + b e(p, y)
+*)
+
+open Snarkette.Mnt6753
+
+(* 
+   Check
+
+   e(p, x + y) = e(p, x) + e(p, y)
+*)
+let check_bilinearity (p : G1.t) (x : G2.t) (y : G2.t) =
+  Fq6.equal
+    (Pairing.reduced_pairing p
+       G2.(x + y))
+    Fq6.(Pairing.reduced_pairing p x * Pairing.reduced_pairing p y)
+
+(* TODO:
+
+   - Sample random pts on E2 using sage
+   - use G2.of_affine_coordinates and copy/paste in to this file.
+   - call this function
+
+   dune exec app/bilinearity-test/bilinearity.exe
+
+   in src directory. *)
+
+let () =
```

Comment:
This statement is causing a type error. To expand a bit, when you write

```ocaml
let PATTERN = EXPRESSION
```

The typechecker works as follows.
1. It infers a type `t` for the pattern `PATTERN`.
2. It checks that the expression `EXPRESSION` has type `t`.

So in this case, it first infers that the pattern `()` has type `unit`, and then it checks that the expression `G2.one` has type unit, which it doesn't, hence the type error.